Packs/TAXIIServer/Integrations/TAXII2Server/TAXII2Server_test.py

```diff
@@ -589,10 +589,29 @@ def test_convert_sco_to_indicator_sdo_with_type_file(mocker):
 
     output = convert_sco_to_indicator_sdo(ioc, xsoar_indicator)
     assert 'file:hashes.' in output.get('pattern', '')
-    assert 'MD5' in output.get('pattern', '')
+    assert 'SHA-1' in output.get('pattern', '')
     assert 'pattern_type' in output.keys()
 
 
+def test_build_sco_object(mocker):
```

Comment:
Since you asked me last time we talked (and I agree) it would be a bit better to use `parameterized`. It will allow each use case to be independent such that if one use case fails the rest will still run, and also it makes it pretty convenient to explain the different cases you are testing in each run. 
And so it is not very different than what you did:
```python
@pytest.mark.parametrize('indicator, sco_indicator',[
    (xsoar_indicators[0], sco_indicators[0])
    (xsoar_indicators[1], sco_indicators[1])
    (xsoar_indicators[2], sco_indicators[2])
])
def test_build_sco_object(indicator, sco_indicator):
    """
        Given
            Case 1: xsoar indicator which is of type A (whatever is special in this case)
            Case 2: xsoar indicator which is of type B
            Case 3: xsoar indicator which is of type C
        When
            Running build_sco_object
        Then
            Case 1: validate that the resulted object has all hashes bla bla
            Case 2: validate that the resulted object does not have hashes bla bla
            Case 3: ...
    """
    output = build_sco_object(indicator["stix_type"], indicator["xsoar_indicator"])
    assert output == sco_indicator
``` 

This is just an example that needs to be fine-tuned (probably fix variable namings and complete docstring) and tested.

Reply:
Thank you!